deployment/keystone/deploy.go

```diff
@@ -137,6 +148,101 @@ func DeployContracts(lggr logger.Logger, e *deployment.Environment, chainSel uin
 	}, nil
 }
 
+// DonInfo is DonCapabilities, but expanded to contain node information
+type DonInfo struct {
+	Name         string
+	Nodes        []Node
+	Capabilities []kcr.CapabilitiesRegistryCapability // every capability is hosted on each node
+}
+
+// TODO: merge with deployment/environment.go Node
+type Node struct {
+	ID           string
+	P2PID        string
+	Name         string
+	PublicKey    *string
+	ChainConfigs []*nodev1.ChainConfig
+}
+
+// TODO: merge with deployment/environment.go NodeInfo, we currently lookup based on p2p_id, and chain-selectors needs non-EVM support
+func NodesFromJD(name string, nodeIDs []string, jd deployment.OffchainClient) ([]Node, error) {
+	// lookup nodes based on p2p_ids
+	var nodes []Node
+	selector := strings.Join(nodeIDs, ",")
+	nodesFromJD, err := jd.ListNodes(context.Background(), &nodev1.ListNodesRequest{
+		Filter: &nodev1.ListNodesRequest_Filter{
+			Enabled: 1,
+			Selectors: []*ptypes.Selector{
+				{
+					Key:   "p2p_id",
+					Op:    ptypes.SelectorOp_IN,
+					Value: pointer.ToString(selector),
```

Comment:
This pointer module seems like overkill (and verbose being pre-generics). In most cases it is not actually simplifying:
```suggestion
					Value: &selector,
```
And in the cases where a helper is simplifying, you can trivially declare one generic func:
```go
func ptr[T any](t T) *T { return &t }
```